tracer/test/Datadog.Trace.Security.Unit.Tests/WafMemoryTests.cs

```diff
@@ -0,0 +1,145 @@
+
+            for (int x = 0; x < 250; x++)
+            {
+                Execute(AddressesConstants.RequestBody, "/.adsensepostnottherenonobook", "security_scanner", "crs-913-120");
+            }
+
+            var current = GetMemory();
+            current.Should().BeLessThanOrEqualTo(baseline + OverheadMargin);
+        }
+
+        [Fact]
+        public void RunMemoryLeakCheck()
+        {
+            var initResult = Waf.Create(WafLibraryInvoker, string.Empty, string.Empty);
+            using var waf = initResult.Waf;
+            waf.Should().NotBeNull();
+
+            var args = new Dictionary<string, object> { { AddressesConstants.RequestBody, "/.adsensepostnottherenonobook" } };
+
+            var baseline = GetMemory();
+
+            for (int x = 0; x < 1000; x++)
+            {
+                using var context = waf.CreateContext();
+                var result = context.Run(args, TimeoutMicroSeconds);
+                result.ReturnCode.Should().Be(ReturnCode.Match);
+                var resultData = JsonConvert.DeserializeObject<WafMatch[]>(result.Data).FirstOrDefault();
+                resultData.Rule.Tags.Type.Should().Be("security_scanner");
+                resultData.Rule.Id.Should().Be("crs-913-120");
+            }
+
+            var current = GetMemory();
+            current.Should().BeLessThanOrEqualTo(baseline + OverheadMargin);
+        }
+
+        [Fact]
+        public void UpdateMemoryLeakCheck()
+        {
+            var initResult = Waf.Create(WafLibraryInvoker, string.Empty, string.Empty);
+            using var waf = initResult.Waf;
+            waf.Should().NotBeNull();
+
+            var baseline = GetMemory();
+
+            bool enabled = false;
+
+            for (int x = 0; x < 1000; x++)
+            {
+                var ruleOverrides = new List<RuleOverride>();
+                var ruleOverride = new RuleOverride { Enabled = enabled, Id = "crs-913-120" };
+                ruleOverrides.Add(ruleOverride);
+                var configurationStatus = new ConfigurationStatus(string.Empty) { RulesOverridesByFile = { ["test"] = ruleOverrides!.ToArray() } };
+                configurationStatus.IncomingUpdateState.WafKeysToApply.Add(ConfigurationStatus.WafRulesOverridesKey);
+                var result = waf!.UpdateWafFromConfigurationStatus(configurationStatus);
+                result.Success.Should().BeTrue();
+
+                Execute(waf, AddressesConstants.RequestBody, "/.adsensepostnottherenonobook", "security_scanner", "crs-913-120", enabled);
+
+                enabled = !enabled;
+            }
+
+            var current = GetMemory();
+            current.Should().BeLessThanOrEqualTo(baseline + OverheadMargin);
+        }
+
+        private long GetMemory()
+        {
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+            var proc = Process.GetCurrentProcess();
+            proc.Refresh();
+            return proc.WorkingSet64;
```

Comment:
I would have though private bytes is a better choice, since this would include all allocated non-shared memory. Working set doesn't include memory that's been swapped out, I think.

Reply:
That was my first guess, but then changed because of [this](https://stackoverflow.com/questions/1984186/what-is-private-bytes-virtual-bytes-working-set)
But I can revert it

Reply:
Wondering what continous profiler is using 🤔